docs/introduction.org

```diff
@@ -0,0 +1,106 @@
+* Introduction
+
+Nowadays, multiple languages have emerged to compete with the well-established C language.
+In fact, C is widely used in many places, learned by a lot of engineers, and used for many purposes.
+C is a powerful tool, /but with great power comes great responsibility/, in fact, memory leaks can easily occur by mistake.
+
+That's why new languages come to find a better way to handle errors and memory.
+The Rust language relies strongly on its borrow checker to avoid safe memory but it comes at a cost: increased code complexity
+
+In another case, the Zig language has a purpose to replace the C language.
+Zig wants to simplify the programming development experience with no hidden control flow and no macro to avoid complex mistakes.
+In comparison with the Rust language, Zig uses explicit memory allocation to prevent memory leaks.
```

Comment:
explicit memory allocation does not prevent memory leaks no ?

Reply:
What do you think now ?

Reply:
mhhhhh, even in you C you do explicit memory allocations (malloc, calloc, ...) I would say that what really help zig to avoid those problemes is having allocators that track memory for you

Reply:
Now it's better ?

Reply:
_In comparison with the Rust language, Zig uses explicit memory allocation to track the allocation and deallocation.
Zig offers different allocators that have different purposes, but some can detect memory leaks ([[file:./allocators.org::#General purpose allocator][General purpose allocator]])._

1. I still dont understand, explicit memory allocations dont allow you by themselves to track allocation and deallocation, allocators do.
2. Maybe I dont understand english very well but your sentence with "In comparison..." is not easy to grasp, I dont understand if Rust does use explicit memory allocation or not. (I would rather use "Contrary to Rust", or "Like rust")

Reply:
In https://doc.rust-lang.org/book/ch04-01-what-is-ownership.html, if find that's a thrid method to allocate memory via ownership. I don't know how to write it correctly.